lib/instance.js

```diff
@@ -640,16 +641,24 @@ Instance.prototype.save = function(options) {
         self.dataValues[createdAtAttr] = values[self.Model.rawAttributes[createdAtAttr].field || createdAtAttr] = self.Model.$getDefaultTimestamp(createdAtAttr) || now;
       }
 
+      var tableName;
+      if (self.$options.$schema !== self.Model.$schema ||
+        self.$options.$schemaDelimiter !== self.Model.$schemaDelimiter) {
+        tableName = self.Model.schema(self.$options.$schema, self.$options.$schemaDelimiter).getTableName(options);
+      } else {
```

Comment:
Yeah I think the issue with trying to do this (at least from what I could tell) is that since Model/$Model are set on the prototype of the instance then changing the schema on one instance will change it on all which causes this test case to fail: https://github.com/sequelize/sequelize/pull/4599/files#diff-7de0096efac48f1ebb1ecedda103ca05R362.